### PR TITLE
Milestone: Dynamically add, delete and edit tasks in the todo list app

### PR DESCRIPTION
**Pull request by Kashif-Saleem-Ghuman**
### In this milestone:

- I have removed the hardcoded array of tasks
- I have implemented javascript functionalities to fetch data from user input
- The data is stored in the local storage
- The data is fetched from the local storage and displayed on UI
- The tasks can be deleted, edited, and marked as completed.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,54 +1,66 @@
 /* eslint-disable linebreak-style */
+// eslint-disable-next-line linebreak-style
+/* eslint-disable no-undef */
+/* eslint-disable linebreak-style */
 /* eslint-disable no-unused-vars */
 /* eslint-disable linebreak-style */
 /* "import/no-unresolved": "off" */
 
 // import _ from 'lodash';
```
Comment (omarsalem7) [inline src/index.js]: - [ ] Kindly avoid disabling linters to make your code follow standards and best practise.

File: src/modules/UI.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable import/prefer-default-export */
+/* eslint-disable linebreak-style */
+/* eslint-disable prefer-const */
+/* eslint-disable linebreak-style */
```
Comment (omarsalem7) [inline src/modules/UI.js]: - [ ] As stated before kindly remove theses lines and fix the linters and you can set up linters locally and run this command `npx eslint . --fix` to fix them.

File: src/modules/delete.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable no-undef */
+/* eslint-disable import/prefer-default-export */
+/* eslint-disable linebreak-style */
```
Comment (omarsalem7) [inline src/modules/delete.js]: - [ ] As stated before avoid disable linters and fix them

File: src/modules/updateStatus.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable no-undef */
+/* eslint-disable linebreak-style */
+/* eslint-disable import/prefer-default-export */
+/* eslint-disable linebreak-style */
```
Comment (omarsalem7) [inline src/modules/updateStatus.js]: - [ ] Remove these also as stated before

File: src/index.js [hunk 5]
```diff
@@ -1,54 +1,66 @@
...
+// window.showMenu = showMenu;
+window.deleteTask = deleteTask;
+
+taskInput.addEventListener('keyup', (e) => {
+  const userTask = taskInput.value.trim();
+  if (e.key === 'Enter' && userTask) {
+    if (!isEditTask) { //  if isEdidTask isn't true
+      todos = !todos ? [] : todos;
+      const taskInfo = { name: userTask, status: 'pending' };
+      todos.push(taskInfo);
```
Comment (omarsalem7) [inline src/index.js]: - [ ] Kindly make sure when adding a new task, consider adding an index for the task and the index set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal 5). Take also into consideration that deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @Kashif-Saleem-Ghuman 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- There is no linter error present ✔️
- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️

- [ ] You [MIT license link](https://github.com/Kashif-Saleem-Ghuman/To-do-list-by-Kashif/tree/day3#:~:text=This%20project%20is%20MIT%20licensed.) go to `404 error page`, kindly fix it and update it with your to make your project professional
- [ ] Also you forget to update your readme info according to the to-do list project instead of an awesome book project.

_Check the comments under the review._

## Optional suggestions

- [ ] [OPTIONAL] It will be better if you choose a descriptive name for your branch like `add-remove-milestone` rather than `day3`

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — CHANGES_REQUESTED** [review]
### Hello @Kashif-Saleem-Ghuman 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
# Required Changes ♻️

- [ ] As the previous reviewer said, your [MIT license link](https://github.com/Kashif-Saleem-Ghuman/To-do-list-by-Kashif/tree/day3#:~:text=This%20project%20is%20MIT%20licensed.) go to 404 error page, kindly fix it and update it with your to make your project professional

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

File: src/index.js (hunk 1 above)
Comment (andersonlebon) [inline src/index.js]: - [ ] As the previous reviewer said, kindly avoid disabling linters to make your code follow standards and best practise. set up linters locally and run this command `npx eslint . --fix`

File: src/modules/UI.js (hunk 2 above)
Comment (andersonlebon) [inline src/modules/UI.js]: - [ ] As the previous reviewer said, kindly fix also this part. set up linters locally and run this command `npx eslint . --fix`

File: src/modules/delete.js (hunk 3 above)
Comment (andersonlebon) [inline src/modules/delete.js]: - [ ] As the previous reviewer said, kindly fix also this part. set up linters locally and run this command `npx eslint . --fix`

File: src/modules/updateStatus.js (hunk 4 above)
Comment (andersonlebon) [inline src/modules/updateStatus.js]: - [ ] As the previous reviewer said, kindly fix also this part. set up linters locally and run this command `npx eslint . --fix`

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @Kashif-Saleem-Ghuman  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request :red_circle: :red_circle: :red_circle: 

You have submitted a project, that has no linter checks, Please make sure that the linter checks are running.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Kashif-Saleem-Ghuman👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️

- [ ] As requested by the previous reviewer instead of disabling the linter in files, Please fix them manually. (Do this in all JS files where you disabled the linter).
  - This command will help you to fix the linter automatically `npx eslint . --fix`

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kashif-Saleem-Ghuman** [comment]
@thecodechaser  I try using that command but those errors don't go away, and were these not just optional requirements?

**Comment by Kashif-Saleem-Ghuman** [comment]
@thecodechaser   i used and the command and these errors don't go away. 
error
)

**Comment by thecodechaser** [comment]
@Kashif-Saleem-Ghuman Fix them manually one by one, Read the error and fix accordingly

**Comment by Kashif-Saleem-Ghuman** [comment]
> @Kashif-Saleem-Ghuman Fix them manually one by one, Read the error and fix accordingly

> The ones that say undefined variables are just linter things because the app is working fine. If a variable is undefined how can the app work?

> Similarly it says clearBtn defined but never used so that is also not fixable, only I can disable it

**Comment by Kashif-Saleem-Ghuman** [comment]
@Kashif-Saleem-Ghuman Fix them manually one by one, Read the error and fix accordingly
@thecodechaser 
The ones that say undefined variables are just linter things because the app is working fine. If a variable is undefined how can the app work?

Similarly it says clearBtn defined but never used so that is also not fixable, only I can disable it

**Comment by thecodechaser** [comment]
@Kashif-Saleem-Ghuman In the screenshot I can see, You are on the `day4` branch, but this PR and code belong to the `day3` branch.
Please confirm this.

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @Kashif-Saleem-Ghuman, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

✔️ You have done a great job working on the requested changes.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**